rasa/core/policies/rule_policy.py

```diff
@@ -227,38 +191,39 @@ def _create_form_unhappy_lookup_from_states(
         lookup = {}
         for states, actions in zip(trackers_as_states, trackers_as_actions):
             action = actions[0]
-            active_form = self._get_active_form_name(states[-1])
+            active_loop = get_active_loop_name(states[-1])
             # even if there are two identical feature keys
-            # their form will be the same
-            # because of `active_form_...` feature
-            if active_form:
-                states = self._modified_states(states)
+            # their loop will be the same
+            if active_loop:
```

Comment:
let's return early to avoid one indentation level
```suggestion
            if not active_loop:
                continue
           # continue rest here
```